### ISSUE TITLE
Server considered installed even when installing fails

### ISSUE THREAD

**Issue by AmeerTaweel** [kabouzeid/nvim-lspinstall#129]
## Environment Info

+ NeoVim version: NVIM v0.6.0-dev+65-g46009499a
+ OS: Ubuntu 21.04
+ Shell: Fish Shell v3.1.2
+ Terminal: Alacritty v0.9.0-dev

## How To Reproduce

+ Start NeoVim, with LSP-Install already installed via Packer
+ System should not have `yarn` installed
+ Run `:LspInstall yaml`
+ The installation should fail since it needs `yarn`
+ Run `:lua for _, server in ipairs(require"lspinstall".installed_servers()) do print(server) end`, which basically prints the supposedly installed servers. You will find yaml there, even though installation failed

This is just an example, if another language server fails to install, the same problem happens.

## Expected Behavior

The server that failed to be installed (in this case `yaml-language-server`) should not be in the list of installed servers.

## Expected Reason

I think that the plugin appends the server it wants to install to the list of installed servers before it runs the actual install, and then it does not check if the install fails or not. But I'm not sure.

**Comment by kabouzeid** [comment]
true, this should be fixed. ie when the exit code is nonzero, then delete the install directory.

**Comment by kabouzeid** [comment]
fixed
